### DISCUSSION TITLE
Confusing SignedDistance calculation in QuadraticSegment::signedDistance

### DISCUSSION THREAD

**Discussion by jkb0o** [Chlumsky/msdfgen#175] [Q&A] [answered]
Hello! I'm trying to port you amazing library to the pure rust (almost brainlessly) and I found confusing part in the SignedDistance calculation:

https://github.com/Chlumsky/msdfgen/blob/eeec8f222d0b06f2636ccb3b9dda3f6c3f00d25c/core/edge-segments.cpp#L183-L188

There are cross if/else conditions. Looks like the first `if` should test for the outer bounds of `0..1` or the second `if` should test against `0/1`. The same is valid for the `QubicSegment` as well.

I not really deep into math, I do not understand everything, so I would appreciate any explanation of this condition.

Thank you for a great job!

**Comment by Chlumsky** [comment, marked as answer]
I'm not sure I understand your point. The first condition checks if `param` is between 0 and 1. Otherwise, the second condition splits the remaining possibilities - `param` lower than 0 or greater than 1. It does so by comparing to 0.5, the value in the middle, but any value between 0 and 1 would work.

**Reply by jkb0o** [reply]
Yeah, now it looks super natural and I wondered why I was confused. Thank you! Sorry for disturbing.
